pyogrio/_io.pyx

```diff
@@ -834,6 +882,9 @@ def ogr_read(
     }
 
     if ogr_dataset != NULL:
+        # TODO: shouldn't this happen in an exception handler?
```

Comment:
No - my understanding is that `GDALDatasetReleaseResultSet` should always be called to release SQL layer.

The check `if ogr_dataset != NULL` can be removed; if we got to this point, it is always non-null.

Reply:
I mean in the finally block of an exception handler, as there is a lot of code in between that can throw an exception...

Reply:
Yes, that's correct - we probably should add one.  In which case, we should keep the `if ogr_dataset != NULL` within it.